File: Golang/806. Number of Lines To Write String.go
func numberOfLines(widths []int, S string) []int {
    total := 0
    lines := 1
    for _, char := range S {
        if total + widths[int(char)-'a'] > 100 {
            lines += 1
            total = widths[int(char)-'a']
        } else {
            total += widths[int(char)-'a']
        }
    }
    return []int{lines, total}
}

File: Python3/806. Number of Lines To Write String.py
class Solution:
    def numberOfLines(self, widths, S):
        """
        :type widths: List[int]
        :type S: str
        :rtype: List[int]
        """
        if not S:
            return [0, 0]
        total = 0
        lines = 1
        for char in S:
            if total + widths[ord(char)-97] > 100:
                lines += 1
                total = widths[ord(char)-97]
            else:
                total += widths[ord(char)-97]
        return [lines, total]
